### ISSUE TITLE
Incompatible inversify version when used with Theia

### ISSUE THREAD

**Issue by planger** [eclipse/sprotty#201]
Sprotty declares the dependency to inversify with [`5.0.1`](https://github.com/eclipse/sprotty/blob/ae671b54aa48f991ca036906bd04ac0ad309ab9f/package.json#L53) while Theia specifies the same dependency with all minor&patch versions higher included: [`^5.0.1`](https://github.com/eclipse-theia/theia/blob/4a3e133bfdab7076dd0290a59f5523c85673d3f2/packages/core/package.json#L36), which currently resolves to version `5.0.5`. This caused the following incompatibility for me:

```
src/browser/diagram-widget-frontend-module.ts(9,26): error TS2345: Argument of type 'import("/home/philip/Temp/theia-sprotty-view/node_modules/inversify/dts/interfaces/interfaces").interfaces.Bind' is not assignable to parameter of type 'import("/home/philip/Temp/theia-sprotty-view/node_modules/@theia/core/node_modules/inversify/dts/interfaces/interfaces").interfaces.Bind'.
  Call signature return types 'BindingToSyntax<any>' and 'BindingToSyntax<T>' are incompatible.
    The types of 'to(...).inSingletonScope().when' are incompatible between these types.
      Type '(constraint: (request: Request) => boolean) => BindingOnSyntax<any>' is not assignable to type '(constraint: (request: Request) => boolean) => BindingOnSyntax<T>'.
        Types of parameters 'constraint' and 'constraint' are incompatible.
          Types of parameters 'request' and 'request' are incompatible.
            Type 'import("/home/philip/Temp/theia-sprotty-view/node_modules/inversify/dts/interfaces/interfaces").interfaces.Request' is not assignable to type 'import("/home/philip/Temp/theia-sprotty-view/node_modules/@theia/core/node_modules/inversify/dts/interfaces/interfaces").interfaces.Request'.
              The types of 'parentContext.container' are incompatible between these types.
                Type 'Container' is missing the following properties from type 'Container': getAllTagged, getAllNamed
```

It can be worked around with selective resolutions in a project's package.json, but it is a bit ugly to tamper with third-party dependency resolutions.

Any objections to change Sprotty's dependency to `^5.0.1` to be inline with Theia so that it resolves to the same version?

**Comment by rrcomtech** [comment]
Is there any way to use this fix already in Sprotty 0.8.0 or will it only be available in 0.10.0 as implied in #202?

**Comment by planger** [comment]
I'm afraid Sprotty currently doesn't backport changes on past releases, given the incubation status of past relases. Depending on your use case and if it isn't too much trouble, I'd rather switch to the `next` version (nightly builds) and pin a specific `next` version to avoid getting all updates as they occur -- that's what we do in most of the projects where we need the the most recent Theia version. If that is all not an option, you could also patch 0.8.0 and publish it yourself.
I hope that still helps!
